src/libraries/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -394,17 +394,13 @@ private async Task DualModeBeginConnect_IPAddressToHost_Fails_Helper(IPAddress c
     [Trait("IPv6", "true")]
     public class DualModeBeginConnectToIPEndPoint : DualModeBase
     {
-        [Fact] // Base case
-        // "The system detected an invalid pointer address in attempting to use a pointer argument in a call"
+        [Fact]
         public void Socket_BeginConnectV4IPEndPointToV4Host_Throws()
         {
             using (Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp))
             {
                 socket.DualMode = false;
-                Assert.Throws<SocketException>(() =>
-                {
-                    socket.BeginConnect(new IPEndPoint(IPAddress.Loopback, UnusedPort), null, null);
-                });
+                Assert.Throws<NotSupportedException>(() => socket.BeginConnect(new IPEndPoint(IPAddress.Loopback, UnusedPort), null, null));
```

Comment:
From my perspective it was a bug; ConnectAsync throws NotSupportedException here, and the SocketException that was being thrown is "The system detected an invalid pointer address in attempting to use a pointer argument in a call"... basically the implementation was flawed and giving the OS bad inputs.

We can mark it as a breaking change if you like, of course.  I don't have strong feelings about that.

Reply:
I'm also unsure. "breaking change" seems like a big hammer, if we add it for all smaller changes, where the chance to break someone is very low, we may end up having a very noisy breaking change list.

It might be reasonable to add a softer, "needs doc update" label to make sure we keep our docs in sync, if smaller changes of this kind are common.

Reply:
> It might be reasonable to add a softer, "needs doc update" label, if smaller changes of this kind are common.

Seems reasonable to me.  @karelz?

Reply:
It's also worth pointing out that this is an API usage exception -- it indicates a bug in the user's code. We have more flexibility in changing these sorts of exceptions because they shouldn't really ever be handled in a try/catch. It's usually more important to be clear about the root issue here (so the user can fix it) that to maintain backward compat.

That's in contrast to the issue in #41585, which is not an API usage issue.

Reply:
I'll make sure to include the concept of "needs doc update" in our documentation plan.

We are planning to make triple slash comments the source of truth. This is one more argument in favor of doing such change, because documentation would be done directly in the PR modifying the source code behavior.

Right now, since the source of truth is dotnet-api-docs, we should make the change in this file:
https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Net.Sockets/Socket.xml